### ISSUE TITLE
fromDecimal throws 'Amount must be an integer'

### ISSUE THREAD

**Issue by tvjg** [davidkalosi/js-money#10]
While using your library, I ran across some unexpected behavior.

> \> require('js-money').fromDecimal(78.13, 'USD');
{ amount: 7813, currency: 'USD' }

> \> require('js-money').fromDecimal(78.43, 'USD');
TypeError: Amount must be an integer

A little [digging into the code](https://github.com/davidkalosi/js-money/blob/d27f023f2e7fa18a60c0bde659528a22ff6c18a0/lib/money.js#L104) revealed that this is just a variant of the floating point rounding errors that I had hoped to avoid in the first place.

> \> 78.13*100
7813

> \> 78.43*100
7843.000000000001

This then [trips the check](https://github.com/davidkalosi/js-money/blob/d27f023f2e7fa18a60c0bde659528a22ff6c18a0/lib/money.js#L75) in the Money constructor which throws the above TypeError. Is there any feasible workaround for this?

**Comment by davidkalosi** [comment]
I really hate javascript numbers :) 
Do you have any suggestion ? we could probably just round the result.

**Comment by davidkalosi** [comment]
I have just merged #9 that should resolve this issue. If you can please test it (version 0.5.1)

thanks
david

**Comment by tvjg** [comment]
Hey David,

The fix actually broke our test environment. Not quite sure how yet. I will do some further testing and report back -- hopefully with a test case.

**Comment by tvjg** [comment]
Looks like the PR did fix the original issue I reported. I'm unclear if the rounding is totally safe for, say, banking purposes, but it works OK for the type of currency conversion I'm doing -- purely for display purposes.

The break in our unit tests was due to the change in the behavior of `toJSON`. I replaced those calls with the `toString` and everything works fine now.

I'll go ahead and close the issue. Thanks for the help.
